IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m

```diff
@@ -0,0 +1,158 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#if !MSID_EXCLUDE_SYSTEMWV
+
+#import "MSIDAuthenticationSession.h"
+
+#import "MSIDWebviewAuthorization.h"
+#import "MSIDWebOAuth2Response.h"
+#import "MSIDTelemetry+Internal.h"
+#import "MSIDTelemetryUIEvent.h"
+#import "MSIDTelemetryEventStrings.h"
+#import "MSIDNotifications.h"
+#import <SafariServices/SafariServices.h>
+
+@implementation MSIDAuthenticationSession
+{
+    API_AVAILABLE(ios(11.0))
+    SFAuthenticationSession *_authSession;
+    
+    NSURL *_startURL;
+    NSString *_callbackURLScheme;
+
+    id<MSIDRequestContext> _context;
+    
+    MSIDWebUICompletionHandler _completionHandler;
+    
+    NSString *_telemetryRequestId;
+    MSIDTelemetryUIEvent *_telemetryEvent;
+}
+
+
+- (instancetype)initWithURL:(NSURL *)url
+          callbackURLScheme:(NSString *)callbackURLScheme
+                    context:(id<MSIDRequestContext>)context
+{
+    self = [super init];
+    if (self)
+    {
+        _startURL = url;
+        _context = context;
+    }
+    
+    return self;
+}
+
+
+- (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
+{
+    if (!completionHandler)
+    {
+        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"CompletionHandler cannot be nil for interactive session.", nil, nil, nil, _context.correlationId, nil);
+        [self notifyEndWebAuthWithURL:nil error:error];
+        completionHandler(nil, error);
```

Comment:
if completion handler is nil, we shouldn't call it :)

Reply:
Done, let's just log here.